### PR TITLE
Add nrm2 Blas function

### PR DESCRIPTION
**Pull request by KhadidjaRabea**
nmr2 blas function

### CODE REVIEW

File: benchmarks/linear_algebra/blas/level1/nrm2/nrm2_generator.cpp [hunk 1]
```diff
@@ -0,0 +1,68 @@
+
```
Comment (rbaghdadi) [inline benchmarks/linear_algebra/blas/level1/nrm2/nrm2_generator.cpp]: Please follow the Tiramisu coding style. This means your code has to look like the rest of Tiramisu code in style. Here no new line is necessary.

File: benchmarks/linear_algebra/blas/level1/nrm2/nrm2_generator.cpp [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+#include "benchmarks.h"
+#include "math.h"
+/*
+  IMPLEMENTATION OF THE NMR2 FUNCTION IN TIRAMISU 
+  
+   out = sqrt(x*'x) 
+
+   where : 
+   x : is a size N vector and 'x is the transpose of x 
+
```
Comment (rbaghdadi) [inline benchmarks/linear_algebra/blas/level1/nrm2/nrm2_generator.cpp]: New line not necessary (as you can see in usual Tiramisu code).

File: benchmarks/linear_algebra/blas/level1/nrm2/nrm2_generator.cpp [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    // -------------------------------------------------------
+    // Layer I
+    // -------------------------------------------------------
+
+    constant NN("N", expr(N));
+
+    var i("i", 0, NN);
+    input x("x", {"i"}, {NN},p_float64);
+    computation S_init("S_init", {}, expr(cast(p_float64,0)));
+    computation S("S", {i},p_float64);
```
Comment (rbaghdadi) [inline benchmarks/linear_algebra/blas/level1/nrm2/nrm2_generator.cpp]: Tiramisu coding style not respected: missing space before p_float64.

File: benchmarks/linear_algebra/blas/level1/nrm2/nrm2_generator.cpp [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    S.set_expression( S(i-1) + x(i)*x(i));
+    computation result("result", {},p_float64);
+    result.set_expression(cast(p_float64,expr(o_sqrt,S(NN),p_float64))); 
+     
+    
+
```
Comment (rbaghdadi) [inline benchmarks/linear_algebra/blas/level1/nrm2/nrm2_generator.cpp]: This code has a lot of issues related to coding style, can you please address them and update your pull request?

Comment (KhadidjaRabea) [reply benchmarks/linear_algebra/blas/level1/nrm2/nrm2_generator.cpp]: Yes of course.

Comment (KhadidjaRabea) [reply benchmarks/linear_algebra/blas/level1/nrm2/nrm2_generator.cpp]: > Your code has merging conflicts. Can you please look at them and fix them?
Salem Sir, 
I've resolved the conflicts and it 's clean now

**Pull request merged**